scripts/irods/upgrade_configuration.py

```diff
@@ -465,7 +465,8 @@ def convert_legacy_configuration_to_json(irods_config):
             'irods_transfer_buffer_size_for_parallel_transfer_in_megabytes': server_config_v2['advanced_settings']['transfer_buffer_size_for_parallel_transfer_in_megabytes'],
             "irods_ssl_verify_server": os.getenv("irodsSSLVerifyServer", 'hostname'),
             "irods_ssl_ca_certificate_file": os.getenv("irodsSSLCACertificateFile", ''),
-            "irods_ssl_ca_certificate_path": os.getenv("irodsSSLCACertificatePath", '')
+            "irods_ssl_ca_certificate_path": os.getenv("irodsSSLCACertificatePath", ''),
+            "irods_connection_pool_refresh_time_in_seconds": int(legacy_irods_environment.get('irodsConnectionPoolRefreshTime', 300))
```

Comment:
Hmm, not sure this is quite right.

I think you want to update this as if you're working on preparing a 4.3.1 release. That means, the upgrade logic needs to apply this change based on the active schema version number.

The problem with what I just said is that the schema version didn't change between 4.3.0 and 4.3.1, meaning there's no place to do this work. You can see what I mean by looking at `schema_version` in version.json.dist.in for each release.

Hmm, need to think about this a little more.

Reply:
Pointers would be appreciated. Would this change then be in `previous_version` or some other dictionary (assuming the change is in a dictionary)?

See: https://github.com/irods/irods/blob/eebc7854dc5ef54eefedf8f0164e2621a02f6554/scripts/irods/upgrade_configuration.py#L617-L627

---

I think I see what you're getting at. The current change would be right if we had it there the first time. Do we have a part of the update code that checks `irods_version` for updates?

Reply:
After some discussion, we've agreed to do the following:
- Remove "(Optional)" from the configs in irods/irods_docs
- Confirm upgrades from iRODS 4.2.x to 4.3.3 do not run into the schema validation issue

We are not going to bump the schema version for 4.3 (remains at v4). We expect iRODS 5 to ship with a schema version of v5.